BSP/STM32F413/Src/BSP_CAN.c

```diff
@@ -125,10 +128,11 @@ uint8_t BSP_CAN_Write(uint32_t id, uint8_t data[8], uint8_t length) {
 uint8_t BSP_CAN_Read(uint32_t *id, uint8_t *data) {
     if(RxFlag){
```

Comment:
Do we need to check for an RxFlag? Is there some other way of determining whether our fifo is full or not?

Reply:
I am setting the flag whenever there is data, not when it is full. The flag will only go away once all the data is gone. It could also go away if 32 pieces of data arrive between the IRQ and CAN_Read.

Reply:
Sorry, I used the wrong word there. I mean that we should be able to determine whether the fifo is non-empty by comparing the head and tail.

Marking the FIFO as empty when it overflows is probably not the behaviour we would want.

Reply:
I think I did this